test/test.c

```diff
@@ -3929,23 +3929,30 @@ static hb_dict_t * PreparePreset(const char *preset_name)
                         hb_str_vlen(anames)))))))))));
 
         hb_dict_t *audio_dict, *last_audio_dict = NULL;
+
         // Add audio dict entries to list if needed
+        // Keep track of if the list was empty, we use this to decide whether or not to apply
+        // settings to all presets
+        int audio_list_was_empty = 0;
+        if (hb_value_array_len(list) == 0)
+        {
+            audio_list_was_empty = 1;
+            hb_value_array_append(list, hb_dict_init());
+        }
+
         for (ii = 0; ii < count; ii++)
         {
             audio_dict = hb_value_array_get(list, ii);
+            
             if (audio_dict == NULL)
             {
-                break;
+                // More settings specified than preset currently supports.
```

Comment:
This breaks all subsequent logic that uses last_audio_dict since the loop will continue setting last_audio_dict to the newly created dict. You can fix by adding `continue;` after `hb_vaule_array_append()`